plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/CabalAdd.hs

```diff
@@ -0,0 +1,272 @@
+                                                                CodeActionKind (CodeActionKind_QuickFix),
+                                                                Diagnostic (..),
+                                                                Null (Null),
+                                                                VersionedTextDocumentIdentifier,
+                                                                WorkspaceEdit,
+                                                                toNormalizedFilePath,
+                                                                type (|?) (InR))
+import           System.Directory                              (doesFileExist,
+                                                                listDirectory)
+import           System.FilePath                               (dropFileName,
+                                                                makeRelative,
+                                                                splitPath,
+                                                                takeExtension,
+                                                                (</>))
+import           Text.PrettyPrint                              (render)
+import           Text.Regex.TDFA
+
+data Log
+  = LogFoundResponsibleCabalFile FilePath
+  | LogCalledCabalAddCodeAction
+  | LogCalledCabalAddCommand CabalAddCommandParams
+  | LogCreatedEdit WorkspaceEdit
+  | LogExecutedCommand
+  deriving (Show)
+
+instance Logger.Pretty Log where
+    pretty = \case
+      LogFoundResponsibleCabalFile fp -> "Located the responsible cabal file at " Logger.<+> Logger.pretty fp
+      LogCalledCabalAddCodeAction -> "The CabalAdd CodeAction is called"
+      LogCalledCabalAddCommand params -> "Called CabalAdd command with:\n" Logger.<+> Logger.pretty params
+      LogCreatedEdit edit -> "Created inplace edit:\n" Logger.<+> Logger.pretty edit
+      LogExecutedCommand -> "Executed CabalAdd command"
+
+cabalAddCommand :: IsString p => p
+cabalAddCommand = "cabalAdd"
+
+data CabalAddCommandParams =
+     CabalAddCommandParams { cabalPath   :: FilePath
+                           , verTxtDocId :: VersionedTextDocumentIdentifier
+                           , buildTarget :: Maybe String
+                           , dependency  :: T.Text
+                           , version     :: Maybe T.Text
+                           }
+  deriving (Generic, Show)
+  deriving anyclass (FromJSON, ToJSON)
+
+instance Logger.Pretty CabalAddCommandParams where
+  pretty CabalAddCommandParams{..} =
+    "CabalAdd parameters:\n" Logger.<+>
+    "| cabal path: " Logger.<+> Logger.pretty cabalPath Logger.<+> "\n" Logger.<+>
+    "| target: " Logger.<+> Logger.pretty buildTarget Logger.<+> "\n" Logger.<+>
+    "| dependendency: " Logger.<+> Logger.pretty dependency Logger.<+> "\n" Logger.<+>
+    "| version: " Logger.<+> Logger.pretty version Logger.<+> "\n"
```

Comment:
In the logs, this is a bit annoying to read :D 

```suggestion
    "CabalAdd parameters:" <+> vcat
      [ "cabal path:" <+> pretty cabalPath
      , "target:" <+> pretty buildTarget
      , "dependendency:" <+> pretty dependency
      , "version:" <+> pretty version
      ]
```

And I recommend to import
```
import Ide.Logger
import qualified Distribution.Pretty                           as Pretty
```